### PR TITLE
Add fallback controllers list to the ControllerInfo

### PR DESCRIPTION
**Pull request by saikishor**
As discussed on 1st April Jazzy meeting, I've created the basic version of adding the fallback controllers list into the controller info and added tests to verify if it is parsing properly or not, we will later continue with the main feature.

This is done as a part of https://github.com/ros-controls/ros2_control/issues/1468

Better wait for #1502 to be merged first as it might have conflicts with this in the spawner.py (or vice versa)

### CODE REVIEW

**Review by christophfroehlich — APPROVED** [review]
LGTM

File: controller_manager/controller_manager/spawner.py [hunk 1]
```diff
@@ -301,6 +310,50 @@ def main(args=None):
                         )
                         return 1
 
+                if not fallback_controllers and param_file:
```
Comment (dhood) [inline controller_manager/controller_manager/spawner.py:313]: might be helpful to document that the cmd line argument will override the param_file value (assuming that's the behaviour you want!) (forgive the driveby review during WG meeting)

Comment (saikishor) [reply controller_manager/controller_manager/spawner.py:313]: Got it! :+1:

File: controller_manager/controller_manager/spawner.py [hunk 2]
```diff
@@ -210,6 +210,14 @@ def main(args=None):
         action="store_true",
         required=False,
     )
+    parser.add_argument(
+        "--fallback_controllers",
+        help="Fallback controllers list are activated as a fallback strategy when the"
+        " spawned controllers fails. When the argument is provided, it takes precedence over"
```
Comment (bmagyar) [inline controller_manager/controller_manager/spawner.py:216]:
```suggestion
        " spawned controllers fail. When the argument is provided, it takes precedence over"
```

**Review by bmagyar — APPROVED** [review]
